src/image_intel.cpp

```diff
@@ -78,22 +80,45 @@ void IntelPlatformsUpdater::reset()
 
 void IntelPlatformsUpdater::doInstall(const fs::path& file)
 {
-    size_t bootaddr = IMAGE_A_ADDR;
-    const char* mtd = "/dev/mtd/image-a";
+    size_t bootaddr = 0;
+    const char* mtd = nullptr;
+
+    const std::string& filename = file.filename();
 
-    if (getBootAddress() == IMAGE_A_ADDR)
+    if (filename == "image-mtd")
+    {
+        mtd = "/dev/mtd0";
+        releaseFlashDrive();
+    }
+    else if (filename == "image-runtime")
+    {
+        bootaddr = IMAGE_A_ADDR;
+        mtd = "/dev/mtd/image-a";
+
+        if (getBootAddress() == IMAGE_A_ADDR)
+        {
+            bootaddr = IMAGE_B_ADDR;
+            mtd = "/dev/mtd/image-b";
+        }
+    }
+    else if (filename == "image-u-boot" && fs::file_size(file) > 0)
     {
-        bootaddr = IMAGE_B_ADDR;
-        mtd = "/dev/mtd/image-b";
+        mtd = "/dev/mtd/u-boot";
     }
 
-    // NOTE: This process may take a lot of time and we want to show the
-    //       progress from original pflash output.
-    printf("Writing %s to %s\n", file.filename().c_str(), mtd);
-    int rc = system(strfmt("flashcp -v %s %s", file.c_str(), mtd).c_str());
-    checkWaitStatus(rc, "");
+    if (mtd)
+    {
+        // NOTE: This process may take a lot of time and we want to show the
+        //       progress from original pflash output.
+        printf("Writing %s to %s\n", file.filename().c_str(), mtd);
+        int rc = system(strfmt("flashcp -v %s %s", file.c_str(), mtd).c_str());
+        checkWaitStatus(rc, "");
+    }
```

Comment:
You don't handle the `else` case in any way. Is that intentional? Don't you want to report an error?

Reply:
The original bundle contains several additional images of zero size. The `image-u-boot` is one of them and should be skipped in silence.

Reply:
Hmm... Seems, the latest bundles have a non empty `image-u-boot`. Probably it was a bug in previous versions of the firmware build.
Let it will be a feature of `fwupdate`. ;)